File: interviewbit-strings-count-and-say.py
# see https://www.interviewbit.com/problems/count-and-say/

class Solution:
    # @param A : integer
    # @return a strings
    def countAndSay(self, A):
        seq = ["1"]
        for i in range(A - 1):
            temp_seq = []
            group = seq[0]
            count = 1
            for digit in seq[1:]:
                if digit == group:
                    count += 1
                else:
                    temp_seq += [str(count), group]
                    group = digit
                    count = 1

            temp_seq += [str(count), group]

            seq = temp_seq[:]

        return "".join(seq)

File: interviewbit-strings-count-and-say.java
public class Solution {
    public String countAndSay(int a) {
        String sequence = "1";
        for (int i = 1; i < a; i++) {
            char current = sequence.charAt(0);
            int count = 1;
            StringBuilder sb = new StringBuilder();
            for (int j = 1; j < sequence.length(); j++) {
                if (current == sequence.charAt(j)) {
                    count++;
                } else {
                    sb.append(count + Character.toString(current));
                    current = sequence.charAt(j);
                    count = 1;
                }
            }
            sb.append(count + Character.toString(current));
            sequence = sb.toString();
        }
        return sequence;
    }
}
